File: java/src/test/java/problems101_200/BinaryTreeLevelOrderTraversalTest.java
package problems101_200;

import concepts.TreeNode;
import java.util.Arrays;
import java.util.stream.Collectors;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class BinaryTreeLevelOrderTraversalTest extends SampledTest {
    @Test
    public void test() {
        for (BinaryTreeLevelOrderTraversal solution : BinaryTreeLevelOrderTraversal.values()) {
            for (BinaryTreeLevelOrderTraversalSample sample
                : getSamples(BinaryTreeLevelOrderTraversalSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.levelOrder(TreeNode.of(sample.input.root)))
                    .isEqualTo(
                        Arrays
                            .stream(sample.output)
                            .map(ints ->
                                Arrays
                                    .stream(ints)
                                    .boxed()
                                    .collect(Collectors.toList())
                            ).collect(Collectors.toList())
                    );
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/BinaryTreeLevelOrderTraversalTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import concepts.TreeNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class BinaryTreeLevelOrderTraversalTest : SampledTest() {
    @Test
    fun test() =
        BinaryTreeLevelOrderTraversal.entries.forEach { solution ->
            getSamples<Array<BinaryTreeLevelOrderTraversalSample>>()
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.levelOrder(TreeNode.of(*input.root)))
                        .isEqualTo(output.map { it.toList() })
                }
        }
}
